components/ws-manager/pkg/manager/monitor.go

```diff
@@ -360,25 +360,12 @@ func actOnPodEvent(ctx context.Context, m actingManager, status *api.WorkspaceSt
 			}
 		}
 
+		// workaround for https://github.com/containerd/containerd/pull/4214 which can prevent pod status
+		// propagation. ws-daemon observes the pods and propagates this state out-of-band via the annotation.
 		_, gone := wso.Pod.Annotations[wsk8s.ContainerIsGoneAnnotation]
 
 		if terminated || gone {
-			// workaround for https://github.com/containerd/containerd/pull/4214 which can prevent pod status
-			// propagation. ws-daemon observes the pods and propagates this state out-of-band via the annotation.
 			go m.finalizeWorkspaceContent(ctx, wso)
-		} else {
-			// add an additional wait time on top of a deletionGracePeriod
-			// to make sure the changes propagate on the data plane.
-			var gracePeriod int64 = 30
```

Comment:
@csweichel Is there a reason why we no longer wait for `terminated` but instead have a timeout here? To me this feels overly brittle, and the `go func... time.Sleep` might cause out-of-order problems if I'm not mistaken..? Might this is even the cause for: https://github.com/gitpod-io/gitpod/issues/5689

Reply:
This might indeed be the cause. The underlying issue that the unmount of the mark mount in ws-daemon happens during content finalization. Without this unmount the workspace container never stops because containerd cannot unmount the rootfs, hence we won't trigger the finalization in the first place.

There are two ways we can break this circle:
1. the solution that's currently implemented where ws-manager forces workspace content finalisation if we've exceeded the grace period
2. a ws-daemon dispatch based method that functions much like the former containerd workaround, except that does the unmount rather than force a workspace stop on the API plane.

We opted for the former because it was easier/less complex to implement. It would seem that it hard other adverse side effects though.

If we merged the change you're proposing here, we'd be re-introducing the "stuck in stopping" issue if ws-daemon restarts.